[docs] Add "Overview" page to Material UI docs (#32643)
mui/material-ui: docs/data/base/getting-started/overview/overview.md

## Introduction

MUI Base is a library of unstyled React UI components.

> It features the same robust engineering as [Material UI](/material-ui/getting-started/overview), but without implementing Material Design.

Reviewer: ```suggestion
It features the same robust engineering as [Material UI](/material-ui/getting-started/overview/), but without implementing Material Design.
```

It includes prebuilt components with production-ready functionality, along with low-level hooks for transferring that functionality to other components.

> With MUI Base, you can rapidly build on top of our foundational components using any styling solution you choose—no need to override any default style engines or settings.

Reviewer 2: ```suggestion
With MUI Base, you can rapidly build on top of our foundational components using any styling solution you choose—no need to override any default style engine or theme.
```

:::info
**Note:** _as of Q2 2022, Base is currently still in alpha._

We're adding new components regularly, and you're welcome to contribute!

> Look for the [`package: base` label](https://github.com/mui/material-ui/labels/package%3A%20base) on open [issues](https://github.com/mui/material-ui/issues) and [pull requests](https://github.com/mui/material-ui/pulls) in the `@mui/material-ui` repository on GitHub to see what other community members are working on, and feel free to submit your own.

Reviewer 2: ```suggestion
Look for the [`package: base` label](https://github.com/mui/material-ui/labels/package%3A%20base) on open [issues](https://github.com/mui/material-ui/issues) and [pull requests](https://github.com/mui/material-ui/pulls) in the `@mui/material-ui` GitHub repository to see what other community members are working on, and feel free to submit your own.
```

Or "repo"?

:::info
**Note:** _as of Q2 2022, Base is currently still in alpha._

We're adding new components regularly, and you're welcome to contribute!

Look for the [`package: base` label](https://github.com/mui/material-ui/labels/package%3A%20base) on open [issues](https://github.com/mui/material-ui/issues) and [pull requests](https://github.com/mui/material-ui/pulls) in the `@mui/material-ui` repository on GitHub to see what other community members are working on, and feel free to submit your own.
:::

## Advantages of MUI Base

- **Ship faster:** MUI Base gives you the foundational building blocks you need to assemble a sleek and sophisticated user interface in a fraction of the time that it would take to do it all from scratch.
- **You own the CSS:** unlike Material UI, which uses emotion as a default style engine, MUI Base has no built-in styling solution.
  This means you have complete control over your app's CSS.
- **Accessibility:** MUI Base components are built with accessibility in mind.
  We do our best to make all components screen reader-friendly, and offer suggestions for optimizing accessibility throughout our documentation.

> ### MUI Base vs. Material UI

Reviewer: Should it be a h2? It's one on the Material UI page.

```suggestion
## MUI Base vs. Material UI
```

It doesn't feel part of "Advantages of MUI Base"

Author: Agreed—I'm not sure why I went with that in the first draft. 👍

### MUI Base vs. Material UI

> MUI Base features many of the same components as [Material UI](/material-ui/getting-started/overview), but _without the Material Design implementation_.

Reviewer: ```suggestion
MUI Base features many of the same components as [Material UI](/material-ui/getting-started/overview/), but _without the Material Design implementation_.
```
